### ISSUE TITLE
Logo link is ignoring settings: use_directory_urls

### ISSUE THREAD

**Issue by paulo-coutinho** [squidfunk/mkdocs-material#2706]
- [x] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

## Description

Hi,

When i use settings:

```
use_directory_urls: false
```

The logo link is "/" and not "index.html".

Can anyone help me?

Thanks.

### Expected behavior

Logo link need be index.html

### Actual behavior

Logo link is "/"

### Steps to reproduce the bug

use settings:

```
use_directory_urls: false
```
### Package versions

* Python: `3.9.5`
* MkDocs: `1.1.2`
* Material: `7.1.5`

### Project configuration

``` yaml
site_name: Ubook SDK
theme:
  name: material
  language: en
  direction: ltr
use_directory_urls: false
lang: en
```

### System information

* OS: macos bigsur
* Browser: chrome

  [1]: https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md

**Comment by squidfunk** [comment]
Thanks for reporting. To help you, we need all information from the issue template (versions, configuration, etc.).

**Comment by paulo-coutinho** [comment]
Hi, done.

**Comment by squidfunk** [comment]
Thanks. This is a MkDocs bug that AFAICT should be fixed in the next version (1.2). We had a workaround in-place but had to remove it because it interfered with versioning, so we had to remove it. See https://github.com/squidfunk/mkdocs-material/commit/5037094b123164f7f7c5fc364c6f8ebc2d06e2f1

You can use the fix in the commit to achieve the prior behavior for the time being.

**Comment by paulo-coutinho** [comment]
Hi,

I will wait for the new release.

Thanks.
